Uuid.php

```diff
@@ -85,9 +85,17 @@ public function isNull(): bool
         return uuid_is_null($this->uuid);
     }
 
-    public function equals(self $other): bool
+    public function equals($other): bool
     {
-        return 0 === uuid_compare($this->uuid, $other->uuid);
+        if ($other instance self) {
+            return 0 === uuid_compare($this->uuid, $other->uuid);
+        }
+        
+        if (is_string($other)) {
```

Comment:
I think accepting strings makes the method less useful as it forces another check to know the type.

I'm good with accepting other types and returning false.

Reply:
Sounds legit

Reply:
don't forget to add a test :)